### PR TITLE
code normalising according to linter remarks

### PR DESCRIPTION
**Pull request by dsuhinin**
What's inside the box:
- fixed all the linter remarks, so now when we run `golangci-lint run` we probably should see nothing.
- we are ready to integrate `golangci-lint run` into build pipeline.

### CODE REVIEW

File: pkg/api/aim/query/clause.go [hunk 1]
```diff
@@ -18,8 +18,10 @@ func (regexp Regexp) Build(builder clause.Builder) {
 	builder.WriteQuoted(regexp.Column)
 	switch regexp.Dialector {
 	case postgres.Dialector{}.Name():
+		//nolint
```
Comment (jgiannuzzi) [inline pkg/api/aim/query/clause.go:21]: I have fixed in a more targeted way in #418

File: pkg/api/aim/encoding/decoder.go [hunk 2]
```diff
@@ -43,6 +43,7 @@ func (d *reader) readField() ([]byte, error) {
 }
 
 // Decode decodes input stream of data into map[string]interface{}.
+// nolint:gocyclo
```
Comment (jgiannuzzi) [inline pkg/api/aim/encoding/decoder.go:46]: we should probably leave a TODO on the `gocyclo` ones to properly fix them — the linter is right that many of them are way too complex

Comment (dsuhinin) [reply pkg/api/aim/encoding/decoder.go:46]: yes yes. I've decided to also added this moment. I have other portion of changes where you will see such a TODO's

File: .golangci.yml [hunk 3]
```diff
@@ -139,7 +144,7 @@ issues:
   # it can be disabled by `exclude-use-default: false`. To creators all
   # excluded by default patterns execute `golangci-lint run --help`
   exclude:
-    - abcdef
+    - declaration of "err" shadows declaration at line
```
Comment (dsuhinin) [inline .golangci.yml:147]: to avoid message related to `err`.

File: pkg/api/aim/query/query.go [hunk 4]
```diff
@@ -408,6 +410,8 @@ func (pq *parsedQuery) parseList(node *ast.List) (any, error) {
 	return list, nil
 }
 
+// nolint:gocyclo
+// TODO:get back and ifx `gocyclo` problem.
```
Comment (suprjinx) [inline pkg/api/aim/query/query.go:414]: ifx -> fix

**Review by suprjinx — DISMISSED** [review]
Tough task -- thanks!

File: pkg/api/aim/query/query.go (hunk 4 above)
Comment (dsuhinin) [reply pkg/api/aim/query/query.go]: done.

**Review by jgiannuzzi — APPROVED** [review]
LGTM for ci.yml and .golangci.yml — I trust that the the rest is fine!

**Review by jgiannuzzi — APPROVED** [review]

**Pull request merged**